### PR TITLE
Chart filters widget to filter chart data by time

### PR DESCRIPTION
**Pull request by ktkization**

### CODE REVIEW

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: This is getting a bit unwieldy with multiple types of data objects being passed to the same command. I'd rather us have one type of data for each command. Perhaps you can introduce some actions for the project selection change and for the date filter change, and for anything else that's needed? It should also clear up the logic quite a bit, at the moment is a bit difficult to follow. Something for a follow up PR I think?

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart:28]: can we also have an `allTime` option? Or perhaps the default option could be used to show all of the data?

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: nit: space between `!= null`

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: nit: space between `!= null`

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: nit: the `case` statements should be indented from the `switch` statement

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: nit: newline needed at end of file, Github's complaining
[you can set your editor to add this automatically so you don't need to always worry about it]

File: webapp/lib/view.dart
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: I see that `filterChartsByPeriod` doesn't actually use whatever parameter is passed into it, so I'd just remove it. It's also not good practice to pass unnamed parameters forwards - if you're using them, they should be named, but it this case you don't need it.

File: webapp/lib/view.dart
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: I'm guessing this is why you added an argument to `filterChartsByPeriod`? You can rewrite this as:
`(_) => filterChartsByPeriod()` which is still very clear and tidy.

File: webapp/lib/view.dart
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: I'd rather not have a default options that doesn't show anything sensible on the screen. As per my other comment, let's turn this into the option to show data for all time.

File: webapp/lib/view.dart
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: leftover debug print?

File: webapp/web/selector.css
Comment (marianamarasoiu) [inline webapp/web/selector.css]: nit: I'd prefer to have the selectors in a list on separate lines, it helps with readability since space has meaning in CSS selectors, and commas look very much like dots if you're not looking very closely.

File: webapp/web/charts.css
Comment (marianamarasoiu) [inline webapp/web/charts.css]: nit: blank newline at end of file

**Review by marianamarasoiu — APPROVED** [review]
LGTM with a few nits and a couple of other comments. thanks @ktkization !

**Comment by ktkization** [comment]
Looking into the requested fixes @marianamarasoiu

File: webapp/web/charts.css
Comment (ktkization) [reply webapp/web/charts.css]: Added newline

File: webapp/web/selector.css
Comment (ktkization) [reply webapp/web/selector.css]: Put it in a separate file

File: webapp/lib/view.dart
Comment (ktkization) [reply webapp/lib/view.dart]: oops. Removed

File: webapp/lib/view.dart
Comment (ktkization) [reply webapp/lib/view.dart]: Implemented an `allTime` option as per your earlier comment

File: webapp/lib/view.dart
Comment (ktkization) [reply webapp/lib/view.dart]: fixed

File: webapp/lib/view.dart
Comment (ktkization) [reply webapp/lib/view.dart]: Noted. Fixed

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart]: Added newline. I'll setup my editor to do this for me

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart]: properly indented it

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart]: Fixed

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart]: Fixed

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart:28]: Add new option `allTime`. It will show all data

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart]: I agree. Created new action for chart filtering. I'll do the same for project selection in a separate PR once I merge this perhaps

**Comment by ktkization** [comment]
@marianamarasoiu lemme know whether I can merge this once you go through the changes I've made

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [inline webapp/lib/controller.dart:226]: is there a typo here? should this be `filterDate` rather than `filterData`?

File: webapp/lib/controller.dart
Comment (marianamarasoiu) [reply webapp/lib/controller.dart]: great, thanks!

**Review by marianamarasoiu — APPROVED** [review]
thanks very much @ktkization - just one new question/typo, after which I think it's good to merge!

File: webapp/lib/controller.dart
Comment (ktkization) [reply webapp/lib/controller.dart:226]: Fixed this. Changes in new [PR](https://github.com/AfricasVoices/nook_watcher/pull/17)

**Pull request merged**
